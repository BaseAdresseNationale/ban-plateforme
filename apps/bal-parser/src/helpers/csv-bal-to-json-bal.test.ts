import fs from 'node:fs';
import { describe, expect, test } from 'vitest';
import csvBalToJsonBal from './csv-bal-to-json-bal.js';

const pathToMockBalCSV = './apps/bal-parser/data-mock/adresses-21286_cocorico.csv';
const mockBalCSV = fs.readFileSync(pathToMockBalCSV, 'utf8');

describe('csvBalToJsonBal', () => {
  test('should convert BAL CSV list into BAL JSON list', async () => {
    const balJSON = csvBalToJsonBal(mockBalCSV);
    expect(balJSON).toMatchSnapshot();
  });
});
