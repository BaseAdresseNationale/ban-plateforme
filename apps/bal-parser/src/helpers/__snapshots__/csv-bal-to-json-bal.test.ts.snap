// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`csvBalToJsonBal > should convert BAL CSV list into BAL JSON list 1`] = `Promise {}`;
